src/app/page.tsx

```diff
@@ -66,16 +85,31 @@ const Home: React.FC<HomeProps> = () => {
       errorState: ErrorState.SERVER_ERROR,
       refetchFunction: refetchTxInfo,
     });
-  }, [isTxInfoError, txInfoError, refetchTxInfo, handleError]);
+    handleError({
+      error: latestBlockInfoError,
+      hasError: isLatestBlockInfoError,
+      errorState: ErrorState.SERVER_ERROR,
+      refetchFunction: refetchLatestBlockInfo,
+    });
+  }, [
+    isTxInfoError,
+    txInfoError,
+    refetchTxInfo,
+    handleError,
+    isLatestBlockInfoError,
+    latestBlockInfoError,
+    refetchLatestBlockInfo,
+  ]);
 
   return (
     <main className={`relative h-full min-h-svh w-full main-app-background`}>
       <Header />
       <div className="container mx-auto flex justify-center p-6">
         <div className="container flex flex-col gap-6">
+          <Stats latestBlockInfo={latestBlockInfo} />
           <SearchBar searchTerm={searchTerm} setSearchTerm={setSearchTerm} />
-          {txInfo && !isLoadingTxInfo ? (
-            <Transaction transaction={txInfo} />
+          {!!searchTerm ? (
```

Comment:
why is `txInfo` changed to `!!searchTerm`?

Reply:
the loading state is handled inside the <Transaction> component. We want the component to render from the moment user types an input, not when the tx info has finished loading

Reply:
ah makes sense. let's marge the PR this time :)